packages/kit/src/runtime/client/client.js

```diff
@@ -1741,7 +1741,7 @@ export function goto(url, opts = {}) {
 		throw new Error('Cannot call goto(...) on the server');
 	}
 
-	url = resolve_url(url);
+	url = new URL(resolve_url(url));
```

Comment:
no big deal but thought maybe it's worth saving creating another instance if one was already created.
```suggestion
  const resolved = resolve_url(url);
  // force new URL for page.url reactivity
  url = url === resolved ? new URL(resolved) : resolved;
```

Reply:
this isn't a hot code path, brevity is a higher priority